CardanoSharp.Wallet/TransactionBuilding/TransactionBodyBuilder.cs

```diff
@@ -134,5 +139,52 @@ public ITransactionBodyBuilder SetMint(ITokenBundleBuilder tokenBuilder)
             _model.Mint = tokenBuilder.Build();
             return this;
         }
+
+        public ITransactionBodyBuilder RemoveFeeFromChange(ulong? fee = null)
+        {
+            if (fee is null)
+                fee = _model.Fee;
+            
+            //get count of change outputs to deduct fee from evenly
+            //note we are selecting only ones that dont have assets
+            //  this is to respect minimum ada required for token bundles
+            IEnumerable<TransactionOutput> changeOutputs;
+            if(_model.TransactionOutputs
+               .Any(x => x.OutputPurpose == OutputPurpose.Change
+                           && (x.Value.MultiAsset is null 
+                               || (x.Value.MultiAsset is not null 
+                                   && !x.Value.MultiAsset.Any()))))
+            {
+                changeOutputs = _model.TransactionOutputs
+                    .Where(x => x.OutputPurpose == OutputPurpose.Change
+                                && (x.Value.MultiAsset is null
+                                    || (x.Value.MultiAsset is not null
+                                        && !x.Value.MultiAsset.Any())));
+            }
+            else
+            {
+                changeOutputs = _model.TransactionOutputs
+                    .Where(x => x.OutputPurpose == OutputPurpose.Change);
+            }
+            
+            ulong feePerChangeOutput = fee.Value / (ulong)changeOutputs.Count();
+            ulong feeRemaining = fee.Value % (ulong)changeOutputs.Count();
+            bool needToApplyRemaining = true;
+            foreach (var o in changeOutputs
+                         .Where(x => x.OutputPurpose == OutputPurpose.Change
+                           && (x.Value.MultiAsset is null 
+                               || (x.Value.MultiAsset is not null 
+                                   && !x.Value.MultiAsset.Any()))))
```

Comment:
sorry @NicholasMaselli, 
should just be `foreach(var o in changeOutputs)`
we dont need all that linq

Reply:
Like this?

```
foreach (var o in changeOutputs)
            {
                if (needToApplyRemaining)
                {
                    o.Value.Coin = o.Value.Coin - feePerChangeOutput - feeRemaining;
                    needToApplyRemaining = false;
                }else 
                    o.Value.Coin = o.Value.Coin - feePerChangeOutput;
            }

```

Reply:
yeah